### PR TITLE
Added useful helper methods

### PR DESCRIPTION
**Pull request by ryangjchandler**
This PR contains some small changes to the methods available on the trait.

I've created some new methods for getting the cache filename and directory so that users can customise it when they need. I've also added a method to invalidate the cache by setting the timestamp to 10000 seconds before the timestamp of the model file. This will cause Sushi to regenerate the cache file.

I also added a config file and service provider so the config can be published if need be, although there are env values that can also be used.

I'll add these bits to the docs if need be.

### CODE REVIEW

**Comment by ryangjchandler** [comment]
Not sure about all of the changed files. Might need to rebase my branch against master.

**Comment by calebporzio** [comment]
Yep, will need to rebase before I can take a look

**Comment by ryangjchandler** [comment]
> Yep, will need to rebase before I can take a look

Seems more like it now...

File: src/Sushi.php [hunk 1]
```diff
@@ -137,4 +137,28 @@ public function usesTimestamps()
...
+    public function invalidateCache()
+    {
+        touch($this->cacheDirectory().'/'.$this->getCacheFileName(), filemtime($this->getModelPath()) - 10000);
+
+        return $this;
+    }
+
+    public function getCacheDirectory()
+    {
+        return $this->cacheDirectory ?? config('sushi.cache-path', storage_path('framework/cache'));
```
Comment (UnSeulT) [inline src/Sushi.php:150]: According to config/sushi.php, should be 'sushi.cache-directory' instead of 'sushi.cache-path'

**Pull request closed without merging**